tiledb/sm/buffer/buffer.h

```diff
@@ -44,18 +43,91 @@ namespace sm {
 
 class ConstBuffer;
 
-/** Enables reading from and writing to a buffer. */
-class Buffer {
+// =======================================================
+//      BufferBase
```

Comment:
NIT: I'd remove these, the classes are already providing a nice divide.

Reply:
I benefit from section breaks. It reduces cognitive load when scanning code.